### PR TITLE
Clarify the documentation of DoFTools::extract_dofs().

### PR DESCRIPTION
**Pull request by bangerth**
As pointed out on the mailing list, the current documentation is incorrect. I took
the liberty to be more descriptive, as I often do :-)

### CODE REVIEW

File: include/deal.II/dofs/dof_tools.h [hunk 1]
```diff
@@ -1300,6 +1295,20 @@ namespace DoFTools
    * @ref GlossComponentMask),
    * then this is equivalent to selecting <em>all</em> vector components
    * corresponding to this non-primitive base element.
+   *
+   * @param[in] dof_handler The DoFHandler whose enumerated degrees of freedom
+   *   are to be filtered by this function.
+   * @param[in] component_mask A mask that states which components you want
+   *   to select. The size of this mask must be compatible with the number of
+   *   components in the FiniteElement used by @p dof_handler. See
```
Comment (jppelteret) [inline include/deal.II/dofs/dof_tools.h]: used by the `@p` dof_handler

File: include/deal.II/dofs/dof_tools.h [hunk 2]
```diff
@@ -1300,6 +1295,20 @@ namespace DoFTools
...
+   *   @ref GlossComponentMask "the glossary entry on component masks"
+   *   for more information.
+   * @param[out] selected_dof A vector that will hold @p true or @p false
```
Comment (jppelteret) [inline include/deal.II/dofs/dof_tools.h]: `@param[out]` selected_dofs

File: include/deal.II/dofs/dof_tools.h [hunk 3]
```diff
@@ -1300,6 +1295,20 @@ namespace DoFTools
...
+   *   values for each degree of freedom depending on whether or not it
+   *   corresponds to a vector component selected by the mask above. The size
+   *   of this array must equal DoFHandler::n_locally_owned_dofs(), which for
+   *   sequential computations of course equals DoFHandler::n_dofs(). Previous
```
Comment (jppelteret) [inline include/deal.II/dofs/dof_tools.h]: Previous -> The previous

File: include/deal.II/dofs/dof_tools.h [hunk 4]
```diff
@@ -1326,13 +1335,19 @@ namespace DoFTools
...
+   * @param[in] block_mask A mask that states which blocks you want
+   *   to select. The size of this mask must be compatible with the number of
+   *   blocks in the FiniteElement used by @p dof_handler. See
+   *   @ref GlossBlockMask "the glossary entry on block masks"
+   *   for more information.
+   * @param[out] selected_dof A vector that will hold @p true or @p false
+   *   values for each degree of freedom depending on whether or not it
+   *   corresponds to a vector block selected by the mask above. The size
+   *   of this array must equal DoFHandler::n_locally_owned_dofs(), which for
+   *   sequential computations of course equals DoFHandler::n_dofs(). Previous
```
Comment (jppelteret) [inline include/deal.II/dofs/dof_tools.h]: Previous -> The previous

File: include/deal.II/dofs/dof_tools.h [hunk 5]
```diff
@@ -1326,13 +1335,19 @@ namespace DoFTools
...
-   * @param block_mask The block mask that describes which blocks to consider
-   * (see
-   * @ref GlossBlockMask)
-   * @param selected_dofs A vector of length DoFHandler::n_dofs() in which
-   * those entries are true that correspond to the selected blocks.
+   * @param[in] dof_handler The DoFHandler whose enumerated degrees of freedom
+   *   are to be filtered by this function.
+   * @param[in] block_mask A mask that states which blocks you want
+   *   to select. The size of this mask must be compatible with the number of
+   *   blocks in the FiniteElement used by @p dof_handler. See
```
Comment (jppelteret) [inline include/deal.II/dofs/dof_tools.h]: used by the `@p` dof_handler

File: include/deal.II/dofs/dof_tools.h [hunk 6]
```diff
@@ -1326,13 +1335,19 @@ namespace DoFTools
...
+   *   @ref GlossBlockMask "the glossary entry on block masks"
+   *   for more information.
+   * @param[out] selected_dof A vector that will hold @p true or @p false
```
Comment (jppelteret) [inline include/deal.II/dofs/dof_tools.h]: `@param[out]` selected_dofs

**Comment by bangerth** [comment]
All fixed.

**Comment by jppelteret** [comment]
Doc fix = no need to bother the tester.

**Pull request merged**